pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -485,9 +498,10 @@ private LLCRealtimeSegmentZKMetadata updateCommittingSegmentZKMetadata(String re
     // TODO Issue 5953 remove the long parsing once metadata is set correctly.
     committingSegmentZKMetadata.setEndOffset(committingSegmentDescriptor.getNextOffset());
     committingSegmentZKMetadata.setStatus(Status.DONE);
-    committingSegmentZKMetadata.setDownloadUrl(URIUtils
-        .constructDownloadUrl(_controllerConf.generateVipUrl(), TableNameBuilder.extractRawTableName(realtimeTableName),
-            segmentName));
+    // If the download url set by the server is a peer download url format with peer scheme, put an empty string in zk
+    // otherwise just use the location in the descriptor.
+    committingSegmentZKMetadata.setDownloadUrl(isPeerURL(committingSegmentDescriptor.getSegmentLocation()) ? ""
```

Comment:
Can we define ""  as `public static string METADATA_URI_FOR_PEER_DOWNLOAD` in `CommonConstants.Segment` and reference the same thing in the server as well, to detect peer download? Once we decide it is going to be `""` all components need to use the same thing, so might as well make a common definition. A change to this will need to be done keeping backward compat in mind, if at all someone wants to change it.

thanks

Reply:
done. Thanks.